src/disc/omegah/OmegahConnManager.cpp

```diff
@@ -0,0 +1,706 @@
+//*****************************************************************//
+//    Albany 3.0:  Copyright 2016 Sandia Corporation               //
+//    This Software is released under the BSD license detailed     //
+//    in the file "license.txt" in the top-level Albany directory  //
+//*****************************************************************//
+
+#include "OmegahConnManager.hpp"
+#include "Albany_config.h"
+
+#include "Panzer_FieldPattern.hpp"
+
+#include <Omega_h_element.hpp> //topological_singular_name
+#include <Omega_h_for.hpp> //parallel_for
+#include <Omega_h_file.hpp> //write_array, write_parallel
+#include <Omega_h_array_ops.hpp> //get_max
+#include <Omega_h_atomics.hpp> //atomic_fetch_add
+#include <Omega_h_int_scan.hpp> //offset_scan
+
+#include <fstream>
+
+namespace Albany {
+
+struct OmegahTetFaceVtx {
```

Comment:
How about moving all these non-conn-mgr structs/functions to a helper file? I think it would make it easier to parse this file...

Reply:
Sure.  Will do.